frontend/src/features/properties/selector/utils.ts

```diff
@@ -1,15 +1,56 @@
 import { IMapProperty } from 'features/properties/selector/models';
 import { compact } from 'lodash';
+import { Api_ResearchFileProperty } from 'models/api/ResearchFile';
 import { pidFormatter } from 'utils';
 
-export const getPropertyIdentifier = (property: IMapProperty) => {
-  if (!!property.pid) {
-    return `PID: ${pidFormatter(property.pid)}`;
-  } else if (!!property.pin) {
-    return `PIN: ${property.pin}`;
-  } else if (!!property.planNumber) {
-    return `Plan #: ${property.planNumber}`;
-  } else {
-    return compact([property.latitude?.toFixed(5), property.longitude?.toFixed(5)]).join(', ');
+export enum NameSourceType {
+  PID = 'PID',
+  PIN = 'PIN',
+  PLAN = 'Plan #',
+  LOCATION = 'Location',
+  RESEARCH = 'Research Name',
+  NONE = 'n/a',
+}
+
+interface PropertyName {
+  label: NameSourceType;
+  value: string;
+}
+
+export const getPropertyName = (property: IMapProperty): PropertyName => {
+  if (property.pid !== undefined && property.pid?.length > 0 && property.pid !== '0') {
+    return { label: NameSourceType.PID, value: pidFormatter(property.pid.toString()) };
+  } else if (property.pin !== undefined && property.pin?.length > 0 && property.pin !== '0') {
+    return { label: NameSourceType.PIN, value: property.pin.toString() };
+  } else if (property.planNumber !== undefined) {
```

Comment:
assuming this is a string (I think it is), shouldn't this also check for non-zero length? also worth mentioning that !!property.pid checks for non-undefined, non-null, and also non-empty.

Reply:
We could, but perhaps that is another conversation about the expectations for a plan number. Maybe empty string is valid? (very odd)